fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/SelectExpression.java

```diff
@@ -626,21 +626,31 @@ public Optional<MatchInfo> adjustMatch(@Nonnull final PartialMatch partialMatch)
     @Nonnull
     @Override
     public PlannerGraph rewriteInternalPlannerGraph(@Nonnull final List<? extends PlannerGraph> childGraphs) {
+        final var predicateString = "WHERE " + getPredicates().stream()
+                .map(Object::toString)
+                .collect(Collectors.joining(" AND "));
+
+        final var abbreviatedPredicateString = predicateString.length() > 30 ? String.format("%02x", predicateString.hashCode()) : predicateString;
```

Comment:
Would it make sense to display the first 30 characters, maybe with `...` at the end, instead of the hash code?